### PR TITLE
update codebase for 1.65 clippy

### PR DESCRIPTION
**Pull request by guswynn**
moving forward

see https://materializeinc.slack.com/archives/CM7ATT65S/p1667586820893679

### CODE REVIEW

**Comment by benesch** [comment]
LGTM modulo codeowner signoff of the various large error variant changes.

**Review by danhhz — COMMENTED** [review]
persist changes LGTM. thanks for the fix!

**Comment by bkirwi** [comment]
The Kafka result boxing seems chill to me!

**Review by bkirwi — APPROVED** [review]

**Pull request merged**
